include/operations/NonUnitaryOperation.hpp

```diff
@@ -103,5 +103,9 @@ namespace qc {
 
         void dumpOpenQASM(std::ostream& of, const RegisterNames& qreg, const RegisterNames& creg) const override;
         void dumpQiskit(std::ostream& of, const RegisterNames& qreg, const RegisterNames& creg, const char* anc_reg_name) const override;
+
+        [[nodiscard]] std::vector<dd::Qubit> getUsedQubits() const override {
+            return getTargets();
```

Comment:
See comment above. This would have to be changed to
```c++
const auto& targets = getTargets();
return {targets.begin(), targets.end()};
```

Reply:
Changed the code as suggested